operators/pkg/controller/elasticsearch/elasticsearch_controller.go

```diff
@@ -141,12 +142,20 @@ func addWatches(c controller.Controller, r *ReconcileElasticsearch) error {
 	if err := c.Watch(&source.Kind{Type: &corev1.Secret{}}, r.dynamicWatches.Secrets); err != nil {
 		return err
 	}
-	if err := r.dynamicWatches.Secrets.AddHandler(&watches.OwnerWatch{
-		EnqueueRequestForOwner: handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &elasticsearchv1alpha1.Elasticsearch{},
+	if err := r.dynamicWatches.Secrets.AddHandlers(
+		&watches.OwnerWatch{
+			EnqueueRequestForOwner: handler.EnqueueRequestForOwner{
+				IsController: true,
+				OwnerType:    &elasticsearchv1alpha1.Elasticsearch{},
+			},
 		},
-	}); err != nil {
+		&watches.FuncMapWatch{
+			Name: "user-secret-watch",
+			EnqueueRequestsFromMapFunc: handler.EnqueueRequestsFromMapFunc{
+				ToRequests: user.NewToRequestsFuncFromClusterNameLabel(),
+			},
```

Comment:
This would be obsolete IIUC if Elasticsearch owned all user secrets (see my other comment further down)